core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -157,13 +157,14 @@ public void start(Listener<RespT> observer, Metadata headers) {
 
     ClientStreamListener listener = new ClientStreamListenerImpl(observer);
     ListenableFuture<ClientTransport> transportFuture = clientTransportProvider.get(callOptions);
+
     if (transportFuture.isDone()) {
```

Comment:
Is the logic from line 161-180 (including `StreamCreationTask`) covered by any test? If not, we probably should add tests to `ClientCallImplTest`.